filters/tee/tee.go

```diff
@@ -236,43 +295,71 @@ func (spec *teeSpec) CreateFilter(config []interface{}) (filters.Filter, error)
 		return nil, filters.ErrInvalidFilterParameters
 	}
 
-	if u, err := url.Parse(backend); err == nil {
-		tee.host = u.Host
-		tee.scheme = u.Scheme
-	} else {
+	u, err := url.Parse(backend)
+	if err != nil {
 		return nil, err
 	}
 
-	if len(config) == 1 {
-		tee.typ = asBackend
-		return &tee, nil
+	var client *net.Client
+	teeClients.mu.Lock()
+	if client, ok = teeClients.store[u.Host]; !ok {
+		client = net.NewClient(net.Options{
```

Comment:
Need to put client into teeClients.store

Reply:
Also would need to cleanup clients in main_test.go similar to what we do in filters/auth package

Reply:
We might also want to make sure that if there is a closed client in the pool we do not use it.
The problem is that if we call `tee.Close()` we close the client but do not remove it from the map.
wdyt?